src/main/core/main.rs

```diff
@@ -32,11 +32,10 @@ pub fn run_shadow<'a>(args: Vec<&'a OsStr>) -> anyhow::Result<()> {
     let options = match CliOptions::try_parse_from(args.clone()) {
         Ok(x) => x,
         Err(e) => {
+            e.print().unwrap();
             if e.use_stderr() {
-                eprint!("{}", e);
                 std::process::exit(1);
```

Comment:
Semi-orthogonal, but could use a comment here why we're choosing the exit-code based on `use_stderr`

Reply:
Added some comments.